File: 프로그래머스/2/68936. 쿼드압축 후 개수 세기/쿼드압축 후 개수 세기.cs
using System;

public class Solution {
    int[] answer = new int[2];
    public int[] solution(int[,] arr) {
        Quad(arr, 0, 0, arr.GetLength(0));
        return answer;
    }
    
    private void Quad(int[,] arr, int x, int y, int size)
    {
        if(Zip(arr, x, y, size, arr[x, y]))
        {
            if(arr[x, y] == 1)
                answer[1]++;
            else answer[0]++;
            return;
        }
        Quad(arr, x, y, size / 2);
        Quad(arr, x, y + size / 2, size / 2);
        Quad(arr, x + size / 2, y, size / 2);
        Quad(arr, x + size / 2, y + size / 2, size / 2);
    }
    
    private bool Zip(int[,] arr, int x, int y, int size, int val)
    {
        for(int i = x; i < size + x; i++)
        {
            for(int j = y; j < size + y; j++)
            {
                if(arr[i, j] != val)
                    return false;
            }
        }
        return true;
    }
}

File: 프로그래머스/2/68936. 쿼드압축 후 개수 세기/쿼드압축 후 개수 세기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> answer = {0, 0};

void quard(int y, int x, int size, vector<vector<int>> &arr)
{
    int temp = arr[y][x];
    for(int i = y; i < y + size; i++)
    {
        for(int j = x; j < x + size; j++)
        {
            if(temp != arr[i][j])
            {
                quard(y, x, size / 2, arr);
                quard(y, x + size / 2, size / 2, arr);
                quard(y + size / 2, x, size / 2, arr);
                quard(y + size / 2, x + size / 2, size / 2, arr);
                return;
            }
        }
    }
    answer[temp]++;
}
vector<int> solution(vector<vector<int>> arr) {
    quard(0, 0, arr.size(), arr);
    return answer;
}